chrome/content/zotero/zoteroPane.js

```diff
@@ -1834,12 +1834,27 @@ var ZoteroPane = new function()
 				Zotero.debug("Items view not available in itemSelected", 2);
 				return false;
 			}
+			let collectionTreeRow = this.getCollectionTreeRow();
+			// I don't think this happens in normal usage, but it can happen during tests
+			if (!collectionTreeRow) {
+				return false;
+			}
 			
 			var selectedItems = this.itemsView.getSelectedItems();
 			
 			// Display buttons at top of item pane depending on context. This needs to run even if the
 			// selection hasn't changed, because the selected items might have been modified.
-			this.updateItemPaneButtons(selectedItems);
+			this.itemPane.data = selectedItems;
+			let viewMode = {
+			isFeedsOrFeed: collectionTreeRow.isFeedsOrFeed(),
+			isDuplicates: collectionTreeRow.isDuplicates(),
+			isPublications: collectionTreeRow.isPublications(),
+			isTrash: collectionTreeRow.isTrash(),
+			rowCount: this.itemsView.rowCount,
+			};
+			this.itemPane.viewMode = viewMode;
```

Comment:
`viewMode` isn't a good name for an object with properties, and it doesn't really seem like we should be reproducing these anyway. Seems like we should just be passing a reference to the `collectionTreeRow` and the `itemsView`, and the calling code can use the existing methods/properties on those interfaces.

`editable` below could *maybe* merit being set again, since that's more an intrinsic state of the pane.

Reply:
I try to prevent passing the `collectionTreeRow` or something that's not part of the item pane, to uncouple these modules. This leaves space for the possibility that we may want to use the item pane somewhere else in the future where there is no `collectionTreeRow`, I don't know.

My point of view here: Being able to directly access another module encourages people to bind them deeper during future maintenance (which is what happened to some parts of our code) and eventually things get mixed up.

Reply:
the `editable` is better than `viewMode`. I will update

Reply:
Anyway, seems like we don't have a better choice here. So let's pass them into itemPane.

Reply:
Yeah, I don't think there's any reason not to pass the objects. We're already passing a whole bunch of their properties. If we used this somewhere where there was no `collectionTreeRow` or `itemsView`, we'd need to mock those properties anyway. We could define and document more limited interfaces with only the properties that are guaranteed to exist, but the ones we're already using are pretty arbitrary, and we (or a plugin) could easily want to use other properties of the `collectionRowTree` in the future.